plugins/wasmedge_rwkv/rwkv.cpp

```diff
@@ -0,0 +1,258 @@
+#include "rwkv.h"
+#include "ggml.h"
+#include "ggml-alloc.h"
+#include "ggml-backend.h"
+#include "ggml-impl.h"
+
+#include "ggml-cpu.h"
+
+#ifdef GGML_USE_CUDA
+#include "ggml-cuda.h"
+#endif
+
+#ifdef GGML_USE_METAL
+#include "ggml-metal.h"
+#endif
+
+#ifdef GGML_USE_BLAS
+#include "ggml-blas.h"
+#endif
+
+#include <string>
+#include <vector>
+#include <cstring>
+#include <cinttypes>
+#include <cmath>
+#include <fstream>
+#include <unordered_map>
+#include <memory>
+#include <utility>
+
+#define _FILE_OFFSET_BITS 64
+// Puts an optional break point, if debug is enabled.
+#define RWKV_MAYBE_BREAK
+
+#include <sys/stat.h>
+
+#if defined(WIN32) || defined(_WIN32) || defined(__WIN32__) || defined(__NT__)
+#    define stat _stat64
+#    define fstat _fstat64
+#    define ftell _ftelli64
+#    define fseek _fseeki64
+#    if !defined(NDEBUG)
+#        include <intrin.h>
+#        define RWKV_MAYBE_BREAK __debugbreak()
+#    endif
+#else
+#    if !defined(__APPLE__)
+#        define ftell ftello
+#        define fseek fseeko
+#    endif
+#endif
+
+static_assert(sizeof(stat::st_size) >= 8, "File offsets should be 64-bit or else rwkv.cpp will not be able to load model files over 2 GB");
+static_assert(sizeof(decltype(ftell(NULL))) >= 8, "File offsets should be 64-bit or else rwkv.cpp will not be able to load model files over 2 GB");
+
+#define RWKV_MAX_NODES 80000
+
+#include "rwkv_error_handling.inc"
+
+#include "rwkv_utilities.inc"
+
+#include "rwkv_file_format.inc"
+
+#include "rwkv_model_loading.inc"
+
+#include "rwkv_operators.inc"
+
+#include "rwkv_graph.inc"
+
+// API function.
+struct rwkv_context * rwkv_init_from_file(const char * file_path, const uint32_t n_threads, const uint32_t n_gpu_layers) {
+    global_last_error = RWKV_ERROR_NONE;
+
+    std::unique_ptr<struct rwkv_context> ctx(new(std::nothrow) struct rwkv_context());
+    RWKV_ASSERT_NULL_MSG(RWKV_ERROR_CTX | RWKV_ERROR_ALLOC, ctx, "Failed to allocate rwkv_context");
+
+    ctx->model = new(std::nothrow) struct rwkv_model();
```

Comment:
```suggestion
    ctx->model = new(std::nothrow) struct rwkv_model();
    RWKV_ASSERT_NULL_MSG(RWKV_ERROR_MODEL | RWKV_ERROR_ALLOC, ctx->model, "Failed to allocate 
}
```
we should probably check allocation before incrementing reference count